### DISCUSSION TITLE
Lesson 12: liquidation modifier on test not clear

### DISCUSSION THREAD

**Discussion by sunnyStefi** [Cyfrin/foundry-full-course-cu#976] [Q&A] [answered]
Hi guys, 
I'm a bit lost inside making my own tests for Lesson 12, **liquidation** section.

I'm trying to replicate a successful liquidation mocking the ETH price going from 1000e8 to 500e8, but I get `ERROR: Arithmetic over/underflow` inside the `_redeemCollateral` method. 

ALICE had deposited 1 ether and minted 500$ when ETH price was 1000$ per ETH. 
I don't understand why now BOB (Liquidator), can redeem the bonus + debt from ALICE's `s_collateralDeposited` mapping (DSCEngine.sol - line 268). 🤌🤌🤌

Here is my modifier **liquidation** inside DSCETest.t.sol, which throws the mentioned error:

```solidity

    modifier liquidation() {
        //AlICE mints when weth == 1000$ 
        vm.startPrank(ALICE);
        ERC20MockWETH(weth).approve(address(engine), COLLATERAL_AMOUNT_ETH);
        engine.depositCollateralAndMint(weth, COLLATERAL_AMOUNT_ETH, SAFE_MINTING_DEXS_USDWEI); //SAFE_MINTING_DEXS_USDWEI is 500e18 USDWEI
        vm.stopPrank();

        //BOB mints when weth == 500$
        int256 plummetingEthPrice = 500e8;
        AggregatorV3Mock(ethUsdPriceFeed).updateAnswer(plummetingEthPrice);
        ERC20MockWETH(weth).mint(LIQUIDATOR, COLLATERAL_AMOUNT_ETH * 2);

        vm.startPrank(LIQUIDATOR);
        ERC20MockWETH(weth).approve(address(engine), COLLATERAL_AMOUNT_ETH * 2); //the weth contract allows the engine to use the minted weth
        engine.depositCollateralAndMint(weth, COLLATERAL_AMOUNT_ETH * 2, SAFE_MINTING_DEXS_USDWEI);
        stablecoin.approve(address(engine), COLLATERAL_AMOUNT_ETH * 2); //the stablecoin contract allows the engine to use the minted dexs

        uint256 actualDebt = 1e18; //ethereum == 500$ gets doubled to 1000$ because the pice halved
        uint256 bonus = ((actualDebt * 10) / 100);
        uint256 redeemedCollateralWithBonus = actualDebt + bonus;
        console.log(redeemedCollateralWithBonus);

        engine._redeemCollateralFrom(weth, redeemedCollateralWithBonus, ALICE, msg.sender);

        // engine.liquidate(weth, ALICE, SAFE_MINTING_DEXS_USDWEI);
        vm.stopPrank();
        _;
    }
```

Thank you for helping out!  ❤️‍🔥❤️‍🔥❤️‍🔥

**Comment by shikhar229169** [comment, marked as answer]
Yes, that's a known issue. 
You can check this out: https://www.codehawks.com/finding/clm81m6ub01jtw9rukyr1h50o

Alice deposited 1 ETH into engine contract but when bob the liquidator comes, he wants to pay the whole Alice's debt which is 500 DSC and bob will be getting $500 worth of eth which for the current moment will be 1 eth along with 10% bonus on 1 eth. So total amount Bob would be eligible for becomes 1.1 ETH, but Alice only deposited 1 eth and as a result of which we get underflow error because we were subtracting 1.1 eth from 1 eth, when we try to redeem that from Alice's deposited weth balance.

**Comment by theirrationalone** [comment]
@shikhar229169 ! Chopped it down already.... BTW if you behold patrick again... He also hinted whilst teaching us the concept.

**Comment by sunnyStefi** [comment]
Yes thank you sooo much! I'm going to go through the whole audit list of bugs now ❤️‍🔥❤️‍🔥❤️‍🔥
